### DISCUSSION TITLE
Use AR functionality of model viewer

### DISCUSSION THREAD

**Discussion by piyushaggarwal02** [google/model-viewer#1896] [Q&A]
I want to view a 3D model in AR using model-viewer. But, I do not want to show my 3D model or any poster on the browser using model viewer. 

I will be customizing my 3D model on the browser using Babylon.js natively (as I had earlier started with this approach) and then I want to show the customized 3D model in AR. There will be just a button, and on clicking this button, a 3D model  should be visible in AR using model viewer. Is this possible? 

And if yes, then how?

**Comment by elalish** [comment]
Interesting; that's not really a use case we had in mind, but it should be possible. You could simply make a small model-viewer element and then make a [custom](https://modelviewer.dev/examples/customizing-ui.html) AR button that completely covers it. The 3D model would still load and render in the background, but it'll only render once if it's not being interacted with, and the shader compilation and such are all shared with the WebXR experience anyway.
